pennylane/plugins/__init__.py

```diff
@@ -11,6 +11,50 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Top level PennyLane module"""
+"""
+Reference plugins
+=================
+
+**Module name:** :mod:`pennylane.plugins`
+
+.. currentmodule:: pennylane.plugins
+
+PennyLane supports a collection of built-in quantum operations and observables,
+including both discrete-variable (DV) operations as used in the qubit model,
+and continuous-variable (CV) operations as used in the qumode model of quantum
+computation.
+
+Here, we provide two reference plugin implementations; one supporting qubit
+operations and observables (:mod:`'default.qubit' <pennylane.plugins.default_qubit>`)
+and  one supporting CV operations and observables
+(:mod:`'default.gaussian' <pennylane.plugins.default_gaussian>`).
+
+These reference plugins provide a very basic built-in qubit and CV circuit
+simulator that can be used with PennyLane without the need for additional
+dependencies. They may also be used in the PennyLane test suite in order
+to verify and test quantum gradient computations.
+
+.. note::
+
+    When writing a plugin device for PennyLane, make sure that your plugin
+    supports as many of the PennyLane built-in operations defined here as possible.
+
+    If the convention differs between the built-in PennyLane operation
+    and the corresponding operation in the targeted framework, ensure that the
+    conversion between the two conventions takes places automatically
+    by the plugin device.
+
+
+Architecture-specific operations
+--------------------------------
+
+.. rst-class:: contents local topic
+
+.. toctree::
+    :maxdepth: 2
+
+    plugins/default_qubit
+    plugins/default_gaussian
+"""
```

Comment:
As part of the TOC refactor, the two reference plugins have been moved to a single TOC entry. As per #271, we should add more user-facing documentation regarding use of the reference plugins to the front-matter.